scripts/check_blocking_packages.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+
+import argparse
+
+import rosdistro
+from rosdistro.dependency_walker import DependencyWalker
+
+index = rosdistro.get_index(rosdistro.get_index_url())
+valid_distro_keys = index.distributions.keys()
+valid_distro_keys.sort()
+
+parser = argparse.ArgumentParser(
+    description='Check for unreleased dependencies of unreleased repositories \
+                 for a new rosdistro.')
+parser.add_argument(
+    '--rosdistro', metavar='ROS_DISTRO', type=str,
+    help='The ROS distribution to check packages for',
+    default=valid_distro_keys[-1])
+
+# If not specified, check for all repositories in the previous distribution
+parser.add_argument(
+    '--repositories',
+    metavar='repositories', type=str, nargs='*',
+    help='Unreleased repositories to check dependencies for')
+
+parser.add_argument(
+    '--depth',
+    metavar='depth', type=int,
+    help='Maxmium depth to crawl the dependency tree')
+
+args = parser.parse_args()
+
+distribution_key = args.rosdistro
+repository_names = args.repositories
+dependency_walker_depth = args.depth
+prev_distribution_key = None
+
+# Find the previous distribution to the current one
+try:
+    i = valid_distro_keys.index(distribution_key)
+except ValueError:
+    print('Distribution key not found in list of valid distributions.')
+    exit(-1)
+if i == 0:
+    print('No previous distribution found.')
+    exit(-1)
+prev_distribution_key = valid_distro_keys[i - 1]
+
+cache = rosdistro.get_distribution_cache(index, distribution_key)
+distro_file = cache.distribution_file
+
+prev_cache = rosdistro.get_distribution_cache(index, prev_distribution_key)
+prev_distribution = rosdistro.get_cached_distribution(
+    index, prev_distribution_key, cache=prev_cache)
+
+prev_distro_file = prev_cache.distribution_file
+
+dependency_walker = DependencyWalker(prev_distribution)
+
+if repository_names is None:
+    # Check missing dependencies for packages that were in the previous
+    # distribution that have not yet been released in the current distribution
+    # Crawl the previous distribution file for previously released packages
+    prev_repository_names = set(
+        prev_distro_file.get_data()['repositories'].keys())
+else:
+    # Check the input arguments and remove entries that were already released.
+    prev_repository_names = set(repository_names)
+
+current_repository_names = set(distro_file.get_data()['repositories'].keys())
+# Print the repositories that will be eliminated from the input
+eliminated_repositories = prev_repository_names.intersection(
+    current_repository_names)
```

Comment:
This check is not sufficient. It needs to check if the repository has a release section as well as that the release section has a version.